src/go/rdctl/cmd/snapshot.go

```diff
@@ -72,64 +73,82 @@ func wrapSnapshotOperation(wrappedFunction cobraFunc) cobraFunc {
 			return err
 		}
 		defer removeBackendLock(appPaths.AppHome)
-
-		connectionInfo, err := config.GetConnectionInfo()
-		if errors.Is(err, os.ErrNotExist) {
-			// If we cannot get connection info from config file (and it
-			// is not specified by the user) then assume main process is
-			// not running.
-			return wrappedFunction(cmd, args)
-		} else if err != nil {
-			return fmt.Errorf("failed to get connection info: %w", err)
-		}
-
-		// Ensure backend is running if the main process is running at all
-		rdClient := client.NewRDClient(connectionInfo)
-		state, err := rdClient.GetBackendState()
-		if errors.Is(err, client.ErrConnectionRefused) {
-			// If we cannot connect to the server, assume that the main
-			// process is not running.
-			return wrappedFunction(cmd, args)
-		} else if err != nil {
-			return fmt.Errorf("failed to get backend state: %w", err)
+		if err := ensureBackendStopped(cmd); err != nil {
+			return err
 		}
-		if state.VMState != "STARTED" && state.VMState != "DISABLED" {
-			return fmt.Errorf("Rancher Desktop must be fully running or fully shut down to do a snapshot-%s action, state is currently %v", cmd.Name(), state.VMState)
+		if err := wrappedFunction(cmd, args); err != nil {
+			factoryreset.DeleteData(appPaths, true)
+			return err
 		}
+		// Note that this does not wait for the backend to be in the
```

Comment:
This comment raises the question if the backend is not `STARTED`, what could be going on?

First, if k8s is disabled, the backend state should be `DISABLED`, not `STARTED`. Is there a race condition (which we maybe shouldn't care about) betwen the backend getting into the `STARTED` state and the snapshot lock getting removed?

I'm just a bit confused here.

Reply:
You're right that the comment doesn't cover the case where k8s is disabled - if k8s is disabled, it would be waiting for the `DISABLED` state. I will update the comment so this confusion doesn't happen again.

I'll reach out to you as to the other confusion, I'm a little confused by your confusion :grin: